cherab/tools/observers/group/base.py

```diff
@@ -83,420 +78,257 @@ def __getitem__(self, item):
             else:
                 raise ValueError("The length of 'samples_per_task' ({}) "
-                                 "mismatches the number of sight-lines ({}).".format(len(value), len(self._sight_lines)))
+                                 "mismatches the number of sight-lines ({}).".format(len(value), len(self._observers)))
         else:
-            for sight_line in self._sight_lines:
-                sight_line.samples_per_task = value
-
-    @property
-    def pipelines(self):
-        # A list of all pipelines connected to each sight-line in the group.
-        return [sight_line.pipelines for sight_line in self._sight_lines]
-
-    def connect_pipelines(self, properties=[(SpectralRadiancePipeline0D, None, None)]):
-        """
-        Connects pipelines of given kinds and names to each sight-line in the group.
-        Connected pipelines are non-accumulating by default.
-
-        :param list properties: 3-tuple list of pipeline properties in order (class, name, filter).
-                                Default is [(SpectralRadiancePipeline0D, None, None)].
-                                The following pipeline classes are supported:
-                                    SpectralRadiacnePipeline0D,
-                                    SpectralPowerPipeline0D,
-                                    RadiacnePipeline0D,
-                                    PowerPipeline0D.
-                                Filters are applied to the mono pipelines only, namely,
-                                PowerPipeline0D or RadiacnePipeline0D. The values provided for spectral
-                                pipelines will be ignored. The filter must be an instance of
-                                SpectralFunction or None.
-
-        """
-
-        for sight_line in self._sight_lines:
-            sight_line.connect_pipelines(properties)
-
+            for observer in self._observers:
+                observer.samples_per_task = value
+    
     def observe(self):
         """
         Starts the observation.
         """
-        for sight_line in self._sight_lines:
+        for sight_line in self._observers:
             sight_line.observe()
-
-    def _get_same_pipelines(self, item):
-        pipelines = []
-        sight_lines = []
-        for sight_line in self._sight_lines:
-            try:
-                pipelines.append(sight_line.get_pipeline(item))
-            except (ValueError, IndexError):
-                continue
-            else:
-                sight_lines.append(sight_line)
-
-        if len(pipelines) == 0:
-            raise ValueError("Pipeline {} was not found for any sight-line in this {}.".format((item, self.__class__.__name__)))
-
-        pipeline_types = set(type(pipeline) for pipeline in pipelines)
-        if len(pipeline_types) > 1:
-            raise ValueError("Pipelines {} have different types for different sight-lines.".format(item))
-
-        return pipelines, sight_lines
```

Comment:
I missed this function in my review. I suggest it should be changed the following way:
```python
    def _get_same_pipelines(self, item):
        pipelines = []
        observers = []
        for observer in self._observers:
            if isinstance(item, int):
                if item < len(observer.pipelines):
                    pipelines.append(observer.pipelines[item])
                    observers.append(observer)
            elif isinstance(item, str):
                matching_pipelines = [pipeline for pipeline in observer.pipelines if pipeline.name == item]
                if len(matching_pipelines) == 1:
                    pipelines.append(matching_pipelines[0])
                    observers.append(observer)

        if len(pipelines) == 0:
            raise ValueError("Pipeline {} was not found for any observer in this {}.".format((item, self.__class__.__name__)))

        pipeline_types = set(type(pipeline) for pipeline in pipelines)
        if len(pipeline_types) > 1:
            raise ValueError("Pipelines {} have different types for different observers.".format(item))

        return pipelines, observers
```

Reply:
If plotting functions are moved outside the base class, this function must be public.

Reply:
I have moved this function into plotting module.